src/protocols/secure_channel/CASEServer.cpp

```diff
@@ -43,20 +43,18 @@ CHIP_ERROR CASEServer::ListenForSessionEstablishment(Messaging::ExchangeManager
     mExchangeManager          = exchangeManager;
     mGroupDataProvider        = responderGroupDataProvider;
 
+    // Setup CASE state machine using the credentials for the current fabric.
+    GetSession().SetGroupDataProvider(mGroupDataProvider);
```

Comment:
The comment doesn't really match. mGroupDataProvider is the data provider for *all* fabrics.

Reply:
Updated as per the suggestion above.